### ISSUE TITLE
Drop Content-Type header for empty request bodies

### ISSUE THREAD

**Issue by MisterMX** [go-openapi/runtime#231]
Similar to #64.

The client always adds a `Content-Type` header to a request. Even if there is no body. Some servers fail to handle the request because they try to parse a body (which is not there).

A workaround as described in https://github.com/go-swagger/go-swagger/pull/1249#issuecomment-736526626 is to set `Content-Type` to `text/plain` which would a string parser allow to read an empty body without failing. However, this does not work in our use case because the API we are using explicitly denies requests with unexpected content types.

Looking at the code, I think the issue is caused by a typo in [this line](https://github.com/go-openapi/runtime/blob/23bff4882b47f57fa7e95edb728109514ce9f1c1/client/request.go#L215):

```golang
	if runtime.CanHaveBody(r.method) && body == nil && r.header.Get(runtime.HeaderContentType) == "" {
		r.header.Set(runtime.HeaderContentType, mediaType)
	}
```

Instead of `body == nil` I guess it should be `body != nil`. Otherwise it would not make sense in my eyes if header should be set for no body. Or am I missing something?

**Comment by youyuanwu** [comment]
`body != nil` seems to be intended. @casualjim  can you confirm?

**Comment by casualjim** [comment]
yeah I agree.  Fixed on v0.23.2
